framework/src/vectorpostprocessors/EigenvalueProstprocessor.C

```diff
@@ -0,0 +1,54 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "EigenvaluePostprocessor.h"
+#include "libmesh/libmesh_config.h"
+
+template<>
+InputParameters validParams<EigenvaluePostprocessor>()
+{
+  InputParameters params = validParams<GeneralVectorPostprocessor>();
+  return params;
+}
+
+EigenvaluePostprocessor::EigenvaluePostprocessor(const InputParameters & parameters) :
+    GeneralVectorPostprocessor(parameters),
+    _eigen_values_real(declareVector("eigen_values_real")),
+    _eigen_values_imag(declareVector("eigen_values_imag")),
+    _nl_eigen(dynamic_cast<NonlinearEigenSystem &>(_fe_problem.getNonlinearSystemBase()))
```

Comment:
This Postprocessor will break if this cast fails. The fact that you are casting to a reference is troubling if it's not guaranteed. You might want to cast to a pointer and then throw an error if this cast fails.